### ISSUE TITLE
CIP-36 test vector CBOR incorrect

### ISSUE THREAD

**Issue by rooooooooob** [cardano-foundation/CIPs#425]
as per `schema.cddl`:
```
delegation = ($voting_pub_key, $weight)

key_registration = {
  1 : [+delegation] / legacy_key_registration,
  2 : $stake_credential,
  3 : $reward_address,
  4 : $nonce,
  ? 5 : $voting_purpose .default 0
}
```

you can see the non-legacy delegation (key 1) is an array of `delegation`, a basic group that isn't array-encoded where it's defined.

However the test vector for the non-legacy format is `a119ef64a50182825820a6a3c0447aeb9cc54cf6422ba32b294e5e1c3ef6d782f2acff4a70694c4d16630182582000588e8e1d18cba576a4d35758069fe94e53f638b6faf7c07b8abd2bc5c5cdee0302582086870efc99c453a873a16492ce87738ec79a0ebd064379a62e2c9cf4e119219e03581de0ae3a0a7aeda4aea522e74e4fe36759fca80789a613a58a4364f6ecef041904d20500` which decoded (using cbor.me) is:
```
{
    61284: {
        1: [
              [h'A6A3C0447AEB9CC54CF6422BA32B294E5E1C3EF6D782F2ACFF4A70694C4D1663', 1],
              [h'00588E8E1D18CBA576A4D35758069FE94E53F638B6FAF7C07B8ABD2BC5C5CDEE', 3]
        ],
        2: h'86870EFC99C453A873A16492CE87738EC79A0EBD064379A62E2C9CF4E119219E',
        3: h'E0AE3A0A7AEDA4AEA522E74E4FE36759FCA80789A613A58A4364F6ECEF',
        4: 1234,
        5: 0
    }
}
```

likewise the final human-readable part of the test vector says (2 element array with 2 element arrays as the elements):
```
{
  61284: {
    1: [['0xa6a3c0447aeb9cc54cf6422ba32b294e5e1c3ef6d782f2acff4a70694c4d1663', 1], ['0x00588e8e1d18cba576a4d35758069fe94e53f638b6faf7c07b8abd2bc5c5cdee', 3]],
    2: '0x1c5d88aa573da97e5a4667e0f7c4a9c6a3d848934c3b0a5b9296b401540f2aef',
    3: '0xe0ae3a0a7aeda4aea522e74e4fe36759fca80789a613a58a4364f6ecef',
    4: 1234,
    5: 0
  },
  61285: {
    1: '0x3aaa2e6b43c0a96e880a7d70df84dffb2a1a17b19d7a99a6ed27b91d499b32027c43acfbf6dff097af7634b2ee38c8039af259b0b6a64316f02b4ffee28a0608'
  }
}
```

According to the `schema.cddl` spec that key should actually be (one 4-element array):
```
        1: [
              h'A6A3C0447AEB9CC54CF6422BA32B294E5E1C3EF6D782F2ACFF4A70694C4D1663',
              1,
              h'00588E8E1D18CBA576A4D35758069FE94E53F638B6FAF7C07B8ABD2BC5C5CDEE',
              3,
        ],
```

The test vectors seem to assume that it was instead defined as `delegation = [$voting_pub_key, $weight]`.

**Comment by rooooooooob** [comment]
Kind of related, but not as big of a deal, is that the CBOR there also has a different stake credential between the human readable final output (`0x1c5d88aa573da97e5a4667e0f7c4a9c6a3d848934c3b0a5b9296b401540f2aef`) and the ones beforehand (`0x86870efc99c453a873a16492ce87738ec79a0ebd064379a62e2c9cf4e119219e`). This is the case for both the legacy and new test vectors.

The legacy delegation also differs between the human readable and CBOR parts. In the CBOR it's the same as the 1st one in the non-legacy delegation array (`0xa6a3c0447aeb9cc54cf6422ba32b294e5e1c3ef6d782f2acff4a70694c4d1663`) instead of the `0x0036ef3e1f0d3f5989e2d155ea54bdb2a72c4c456ccb959af4c94868f473f5a0` in the human-readable parts (both intermediate and final).

**Comment by rphair** [comment]
@rooooooooob @Ryun1 perhaps while CIP-0036 is under scrutiny from https://github.com/cardano-foundation/CIPs/pull/373 we could have a corrected CBOR submitted as a PR?  I'll watch out for this & get it classified as a "correction" immediately so we can put the change through quickly.

**Comment by Ryun1** [comment]
@rphair 

Yes, this is a good spot from @rooooooooob, this is a mistake in the CDDL - I have checked this with Catalyst and one of the authors; @kevinhammond. 

Tomorrow I will create a PR with the correction. Thanks for the heads up.

**Comment by Ryun1** [comment]
@rphair @rooooooooob 

Please see, #426.

**Comment by rphair** [comment]
thanks @rooooooooob @Ryun1 @kevinhammond ... closing this so further review can continue in https://github.com/cardano-foundation/CIPs/pull/426.
